src/Combinatorics/Matroids/ChowRings.jl

```diff
@@ -0,0 +1,211 @@
+Tom Braden, June Huh, et. al.
+
+# Examples
+The following computes the augmented chow ring of the Fano matroid.
+```jldoctest
+julia> M = fano_matroid();
+
+julia> R = augmented_chow_ring(M);
+```
+"""
+function augmented_chow_ring(M::Matroid)
+        Flats = flats(M)
+        sizeFlats = length(Flats)
+        n = length(M.groundset)
+
+        if(!is_loopless(M))
+                throw("Matroid has loops")
+        end
+        if(sizeFlats<2)
+                throw("Matroid has too few flats")
+        end
+        proper_flats = Flats[1:sizeFlats-1]
+
+        element_var_names = [string("y_", S) for S in groundset(M)]
+        flat_var_names = [replace(string("x_",S), "["=>"{", "]"=>"}", ", "=>",") for S in proper_flats]
+        flat_var_names[1] = "x_{}" # Override "x_Any{}"
+        var_names = vcat(element_var_names, flat_var_names)
+        s = length(var_names)
+
+        ring, vars = GradedPolynomialRing(QQ, var_names)
+        element_vars = vars[1:n]
+        flat_vars = vars[n+1:s]
+
+        I = augmented_linear_relations(ring, proper_flats, element_vars, flat_vars, M)
+        J = augmented_quadratic_relations(ring, proper_flats, element_vars, flat_vars, M)
+
+        chow_modulus = ideal(ring, vcat(I, J))
+        chow_ring, projection = quo(ring, chow_modulus)
+
+        return chow_ring
+end
+
+function augmented_linear_relations(ring, proper_flats, element_vars, flat_vars, M)
+        n = length(M.groundset)
+
+        relations = Array{MPolyElem_dec{fmpq, fmpq_mpoly}}(undef, n)
+        i = 1
+        for element in groundset(M)
+                relations[i] = element_vars[i]
+                j = 1
+                for proper_flat in proper_flats
+                        if !(element in proper_flat)
+                                relations[i] -= flat_vars[j]
+                        end
+                        j += 1
+                end
+                i += 1
+        end
+        return relations
+end
+
+function augmented_quadratic_relations(ring, proper_flats, element_vars, flat_vars, M)
+        incomparable_polynomials = quadratic_relations(ring, proper_flats, flat_vars)
+        xy_polynomials = Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}()
+
+        i = 1
+        for element in groundset(M)
+                j = 1
+                for proper_flat in proper_flats
+                        if !(element in proper_flat)
+                                push!(xy_polynomials, element_vars[i] * flat_vars[j])
+                        end
+                        j += 1
+                end
+                i += 1
+        end
+
+        return vcat(incomparable_polynomials, xy_polynomials)
+end
+
+"""
+A helper function to select indicies of a vector that do `include` elments of a given set and `exclude` anothers
+"""
+function select(include::Union{AbstractVector,Set},exclude::Union{AbstractVector,Set},set::Union{AbstractVector,Set})
+        all = []
+        for e in set
+                all = union(all,e)
+        end
```

Comment:
This looks weird. Is it the same as writing `all = union(set...)`?

Reply:
It is, I didn't knew this syntax